docs/001-mana_proposal.md

```diff
@@ -0,0 +1,538 @@
+Then, the appropriate node is located in `Base Mana Vector`, and mana is pledged to its `BaseMana`.
+`Events` are essential to study what happens within the module from the outside.
+
+Note, that `revokeBaseMana1` accesses the mana entry of the nodes within `Base Mana Vector`, therefore all values are
+updated with respect to `t`. Notice the two branches after the condition. When `Base Mana` values had been updated before
+the transaction's timestamp, a regular update is carried out. However, if `t` is older, than the transaction timestamp,
+an update in the "past" is carried out and values are updated up to `LastUpdated`.
+```go
+func (bm *BaseMana) revokeBaseMana1(amount float64, t time.Time) {
+	if t.After(bm.LastUpdated) {
+		// regular update
+		n := t.Sub(bm.LastUpdated)
+		// first, update EBM1, BM2 and EBM2 until `t`
+		bm.updateEBM1(n)
+		bm.updateBM2(n)
+		bm.updateEBM2(n)
+
+		bm.LastUpdated = t
+		// revoke BM1 at `t`
+		bm.BaseMana1 -= amount
+	} else {
+		// update in past
+		n := bm.LastUpdated.Sub(t)
+		// revoke BM1 at `t`
+		bm.BaseMana1 -= amount
+		// update EBM1 to `bm.LastUpdated`
+		bm.EffectiveBaseMana1 -= amount*(1-math.Pow(math.E,-EMA_coeff_1*n))
+	}
+}
+```
+The same regular and past update scheme is applied to pledging mana too:
+```go
+func (bm *BaseMana) pledgeAndUpdate(tx *transaction) (bm1Pledged int, bm2Pledged int){
+	t := tx.timestamp
+	bm1Pledged = sum_balance(tx.inputs)
+
+	if t.After(bm.LastUpdated) {
+		// regular update
+		n := t.Sub(bm.LastUpdated)
+		// first, update EBM1, BM2 and EBM2 until `t`
+		bm.updateEBM1(n)
+		bm.updateBM2(n)
+		bm.updateEBM2(n)
+		bm.LastUpdated = t
+		bm.BaseMana1 += bm1Pledged
+		// pending mana awarded, need to see how long funds sat
+		for input := range tx.inputs {
+			// search for the timestamp of the UTXO that generated the input
+			t_inp := LoadTxTimestampFromOutputID(input)
+			bm2Add := 1 / decay * input.balance * (1 - math.Pow(math.E, -decay*(t-t_inp)))
+			bm.BaseMana2 += bm2Add
+			bm2Pledged += bm2Add
+		}
+	} else {
+		// past update
+		n := bm.LastUpdated.Sub(t)
+		// update BM1 and BM2 at `t`
+		bm.BaseMana1 += bm1Pledged
+		oldMana2 = bm.BaseMana2
+		for input := range tx.inputs {
+			// search for the timestamp of the UTXO that generated the input
+			t_inp := LoadTxTimestampFromOutputID(input)
+			bm2Add := 1/decay * input.balance *(1-math.Pow( math.E,-decay*(t-t_inp) ) ) * math.Pow(math.E, -decay*n)
+			bm.BaseMana2 += bm2Add
+			bm2Pledged += bm2Add
+		}
+		// update EBM1 and EBM2 to `bm.LastUpdated`
+		bm.EffectiveBaseMana1 += amount*(1-math.Pow(math.E,-EMA_coeff_1*n))
+		bm.EffectiveBaseMana2 += (bm.BaseMana2-oldMana2)*EMA_coeff_2*(math.Pow(math.E,-decay*n)-math.Pow(math.E,-EMA_coeff_2*n))/(EMA_coeff_2-decay)
+	}
+	return bm1Pledged, bm2Pledged
+}
+```
+
+```go
+func (bm *BaseMana) updateEBM1(n time.Duration) {
+    bm.EffectiveBaseMana1 = math.Pow(1-EMA_coeff_1, n) * bm.EffectiveBaseMana1 +
```

Comment:
There is an inconsistency between the formulas (my fault, not yours). This should be: 

bm.EffectiveBaseMana1 = EMA_coeff_1 * math.Pow(math.E, -EMA_coeff_1 * n) * bm.EffectiveBaseMana1 +  (1-math.Pow(math.E, -EMA_coeff_1 * n)) * bm.BaseMana1

Reply:
updated now.